packages/ActionBar/src/hooks/useColumnsArrangement/useColumnsArrangement.js

```diff
@@ -31,23 +32,71 @@ function updateHiddenColumnIdFromLocalStorage(localStoragePrefix, columnId, isVi
   }
 }
 
+function hasDifferences(originalArray, newArray) {
+  if (originalArray.length !== newArray.length) return true;
+
+  return difference(originalArray, newArray).length > 0 || difference(newArray, originalArray).length > 0;
```

Comment:
I don't know if thie early performance or not but have you consider 

JSON.stringify(orginalArray) !== JSON.stringify(newArray)

for the size of data we are storing it might end be fast enough and easier to compare?

Reply:
Good point! we don't need to care about the items order in the array though, let me take a look if there's a better way